tests/contrib/urllib3/test_urllib3.py

```diff
@@ -0,0 +1,392 @@
+
+    def test_traced_connection_from_url(self):
+        """Tests tracing from ``connection_from_url`` is set up"""
+        conn = urllib3.connectionpool.connection_from_url(URL_200)
+        resp = conn.request("GET", "/")
+        assert resp.status == 200
+        assert len(self.tracer.writer.pop()) == 1
+
+    def test_resource_path(self):
+        """Tests that a successful request tags a single span with the URL"""
+        resp = self.http.request("GET", URL_200)
+        assert resp.status == 200
+        spans = self.tracer.writer.pop()
+        assert len(spans) == 1
+        s = spans[0]
+        assert s.get_tag("http.url") == URL_200
+
+    def test_tracer_disabled(self):
+        """Tests a disabled tracer produces no spans on request"""
+        self.tracer.enabled = False
+        out = self.http.request("GET", URL_200)
+        assert out.status == 200
+        spans = self.tracer.writer.pop()
+        assert len(spans) == 0
+
+    def test_args_kwargs(self):
+        """
+        Test that args are kwargs are correctly inferred from the target function's
+        signature.
+
+        The args/kwargs used in the integration are:
+            - method (idx 0)
+            - url (idx 1)
+            - headers (idx 3)
+        """
+        inputs = [
+            (("POST", URL_200, b"payload", {"accept": "*"}), {}),
+            (("POST", URL_200, b"payload"), {"headers": {"accept": "*"}}),
+            (("POST", URL_200), {"headers": {"accept": "*"}}),
+            (("POST",), {"url": URL_200, "headers": {"accept": "*"}}),
+            ((), {"method": "POST", "url": URL_200, "headers": {"accept": "*"}}),
+        ]
+
+        for args, kwargs in inputs:
+
+            with self.override_config("urllib3", {}):
+                config.urllib3.http.trace_headers(["accept"])
+                pool = urllib3.connectionpool.HTTPConnectionPool(SOCKET)
+                out = pool.urlopen(*args, **kwargs)
+            assert out.status == 200
+            spans = self.tracer.writer.pop()
+            assert len(spans) == 1
+            s = spans[0]
+            assert s.get_tag(http.METHOD) == "POST"
+            assert s.get_tag(http.STATUS_CODE) == "200"
+            assert s.get_tag(http.URL) == URL_200
+            assert s.get_tag("http.request.headers.accept") == "*"
+
+    def test_untraced_request(self):
+        """Disabling tracing with unpatch should submit no spans"""
+        # Assumes patching is done in the setUp of the test
+        unpatch()
+
+        out = self.http.request("GET", URL_200)
+        assert out.status == 200
+        spans = self.tracer.writer.pop()
+        assert len(spans) == 0
+
+    def test_double_patch(self):
+        """Ensure that double patch doesn't duplicate instrumentation"""
+        patch()
+        connpool = urllib3.connectionpool.HTTPConnectionPool(SOCKET)
+        setattr(connpool, "datadog_tracer", self.tracer)
+
+        out = connpool.urlopen("GET", URL_200)
```

Comment:
```suggestion
        out = connpool.request("GET", URL_200)
```

Reply:
Same case as above, this test is checking a double patch on the instrumented class, which is `HTTPConnectionPool` and the patched function on that class is `urlopen`.